docs/explanation/parts.rst

```diff
@@ -0,0 +1,263 @@
+using *plugins*, each part definition will use the ``plugin`` key.
+
+The available plugins are provided as modules in the `craft_parts/plugins`
+directory -- see :py:mod:`craft_parts.plugins`. These aim to simplify the
+process of building source code written in a variety of programming languages
+using appropriate build systems, libraries and frameworks. If a plugin is not
+available for a particular combination of these attributes, a custom plugin
+can be created or a basic plugin can be used to manually specify the build
+actions to be taken, using the ``override-build`` key.
+
+When a plugin is used, it exposes additional keys that can be used to
+define behaviour that is specific to the type of project that the plugin
+supports. For example, the :py:mod:`cmake plugin <craft_parts.plugins.cmake_plugin>` provides the ``cmake-parameters`` and
+``cmake-generator`` keys that can be used to configure how :command:`cmake`
+is used in the build process.
+
+The build process can be further customised with the ``build-environment``,
+``build-attributes`` and ``override-build`` keys.
+
+The ``build-environment`` key defines assignments to shell environment variables in the build environment, specified as a list of key-value pairs.
+In YAML format, this can take either of the following forms:
+
+.. code:: yaml
+
+   build-environment: [MESSAGE: "Hello world", NAME: "Craft Parts"]
+
+   build-environment:
+     - MESSAGE: "Hello world"
+     - NAME: "Craft Parts"
+
+The ``build-attributes`` key allows a number of standard customisations to be
+applied to the build. Some of these are used to address issues that occur in
+specific situations; others, such as ``debug`` are generally useful.
+
+By default, Craft Parts builds independent parts in parallel. This can be
+disabled by setting the ``disable-parallel`` key to ``True``.
+
+The ``override-build`` key is used to override the build process provided by
+a plugin, and it can be used to replace or extend it.
+
+Build products
+~~~~~~~~~~~~~~
+
+After a part has been built in the *build* step, the build products are
+prepared in three further steps.
+
+If the ``organize`` key is specified, it is used to move files
+specify a list of mappings between files
+in the part key-value pairs
+
+   * How to manage the different steps in the build process
+
+    override-stage string
+    override-prime string
+
+
+
+   * Filesets
+
+ * How artifacts are exported/installed
+
+
+
+
+    stage-snaps ref: #/definitions/grammar-array
+    stage-packages ref: #/definitions/grammar-array
+    organize
+    filesets
+    stage array with at least 1 unique item(s) of string
+    prime array with at least 1 unique item(s) of string
+
+..    parse-info array with at least 1 unique item(s) of string
+
+.. _build-order:
+
+Defining the build order
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, when more than one part is specified, they are built in alphabetical order unless there are dependencies between parts.
```

Comment:
I would focus on the need to explicitly define part dependencies (using `after:`) when a part needs another part to be staged before executing its build step, instead of mentioning that the default order is alphabetical (it is, but it shouldn't matter for the final user if the parts are independent).